src/VisualStudio/Core/Def/Implementation/Extensions/DocumentExtensions.cs

```diff
@@ -59,5 +60,15 @@ internal static class DocumentExtensions
 
             return codeBlocks;
         }
+
+        public async static Task<TextSpan> GetSpanFromPositionAsync(this Document document, int startLine, int startColumn, int endLine, int endColumn, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Thanks, I will remove this method and use the existing method on DiagnosticData.